rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+import logging
+import json
+import re
+
+import boto3
+
+from rapid.lib.constants import Constants
+from rapid.lib.framework.injectable import Injectable
+from rapid.lib.work_request import WorkRequest
+from rapid.master.master_configuration import MasterConfiguration
+from rapid.workflow.action_instances_service import ActionInstanceService
+from rapid.workflow.queue_handlers.container_handlers.container_handler import ContainerHandler
+from rapid.workflow.queue_handlers.container_handlers.ecs_configuration import ECSConfiguration
+from rapid.workflow.queue_handlers.queue_handler_constants import register_queue_handler
+
+logger = logging.getLogger('rapid')
+
+
+@register_queue_handler
+class ECSQueueHandler(ContainerHandler, Injectable):
+    __injectables__ = {'rapid_config': None, 'action_instance_service': ActionInstanceService}
+    @property
+    def container_identifier(self):
+        return 'ecs'
+
+    def __init__(self, rapid_config, action_instance_service):
+        """
+        Parameters
+        ----------
+        rapid_config: MasterConfiguration
+        action_instance_service: ActionInstanceService
+        """
+        super(ECSQueueHandler, self).__init__(rapid_config)
+        self.action_instance_service = action_instance_service
+        self._ecs_client = None
+        self._ecs_configuration = None  # type: ECSConfiguration or None
+
+    def process_work_request(self, work_request, clients):
+        # type: (WorkRequest, list) -> bool
+        # 1. Get the Task Definition
+        task_definition = self._get_task_definition(work_request)
+
+        # 2. Set the status to INProgress
+        self._set_task_status(work_request.action_instance_id, Constants.STATUS_INPROGRESS, start_date=datetime.datetime.utcnow())
+
+        # 3. Run the task
+        (status_id, assigned_to) = self._run_task(task_definition)
+
+        # 4. Report the status
+        self._set_task_status(work_request.action_instance_id, status_id, assigned_to)
+        return True
+
+    def process_action_instance(self, action_instance, clients):
+        # TODO - Still to do for handling long running requests.
```

Comment:
chatted with MB on this. He's still not sure how he wants to handle these checks into the action instance, and is okay with this TODO going up.